cobspec/src/main/java/com/saralein/cobspec/Main.java

```diff
@@ -65,4 +46,34 @@ public static void main(String[] args) {
 
         return new ArgsValidation(validators).validate(args);
     }
+
+    private static void createServer(Path root, int port, Logger logger, Path logTxt) {
+        FormStore formStore = new FormStore();
+        FormBody formBody = new FormBody();
+        FormModification formModification = new FormModification();
+        List<String> authRoutes = new ArrayList<String>(){{
+            add("/logs");
+        }};
+
+        new HttpServer(logger)
+                .addStatic(root)
+                .router(new Routes()
+                        .get("/redirect", new RedirectController())
+                        .get("/form", new FormGetController(formStore, formBody))
+                        .post("/form", new FormPostController(formStore, formBody, formModification))
+                        .put("/form", new FormPutController(formStore, formBody, formModification))
+                        .delete("/form", new FormDeleteController(formStore))
+                        .options("/method_options", new OptionsController(Methods.allowNonDestructiveMethods()))
+                        .get("/method_options", new DefaultController())
+                        .put("/method_options", new DefaultController())
+                        .post("/method_options", new DefaultController())
+                        .head("/method_options", new DefaultController())
+                        .options("/method_options2", new OptionsController(Methods.allowGetAndOptions()))
+                        .get("/method_options2", new DefaultController())
+                        .get("/tea", new DefaultController())
+                        .get("/coffee", new CoffeeController())
+                        .get("/logs", new LogController(logTxt)))
+                .use(new AuthMiddleware("admin", "hunter2", "ServerCity", authRoutes))
```

Comment:
I feel like this can be simpler. In order to use this, we have to duplicate the routes between the middleware and the routes. (in this case `"/logs"`). How can we rethink this to eliminate that duplication?